batch_benchmark_test.go

```diff
@@ -0,0 +1,126 @@
+package bolt_test
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+	"sync"
+	"testing"
+
+	"github.com/boltdb/bolt"
+)
+
+func BenchmarkDBBatchAutomatic(b *testing.B) {
+	db := NewTestDB()
+	defer db.Close()
+	db.MustCreateBucket([]byte("bench"))
+
+	h := fnv.New32a()
+	buf := make([]byte, 4)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		start := make(chan struct{})
+		var wg sync.WaitGroup
+
+		for round := 0; round < 1000; round++ {
+			wg.Add(1)
+
```

Comment:
Minor nitpick: This function has a blank line here while the other two that are virtual identical do not.